packages/components/src/components/modal/modal-styles.ts

```diff
@@ -106,20 +111,29 @@ export const getComponentCss = (
             content: '""',
             position: 'fixed',
             ...getInsetJssStyle(),
-            background: `${darkThemeBackgroundColor}e6`, // e6 = 0.9 alpha
+            background: themeDarkBackgroundShading,
+            ...frostedGlassStyle,
           },
         }),
         overflowY: 'auto', // overrideable
       },
       '::slotted': addImportantToEachRule({
-        ...getSlottedJssStyle(32, hasHeader),
-        [mediaQueryM]: getSlottedJssStyle(40, hasHeader),
-        [mediaQueryXxl]: getSlottedJssStyle(64, hasHeader),
+        ...mergeDeep(
+          getSlottedJssStyle(32, hasHeader, disableCloseButton),
```

Comment:
Nested object composition seems unnecessary in front of `mergeDeep`.
Also slotted jss might be confusing since this usually refers to light dom styles that are injected into the head.

Reply:
Nested object comp is fixed. But what do you mean with "confusing slotted CSS"? We need this to react to the border-radius in relation to the fullscreen mode (with no border radius).

Reply:
Slotted css refers to the styles that are injected into the head that look like `p-text a { ... }`.
But in this context it is about the `:slotted` selector as far as I can tell which are styles that are in the shadow dom.

Reply:
Ah, ok, but I didn't created this function name ;-) But I can rename it if you want.

Reply:
Another note: It seems that `...mergeDeep` with nested object comp is needed. Otherwise some selectors get stripped off.